src/Simulation/QsharpCore/Intrinsic.qs

```diff
@@ -657,24 +658,26 @@ namespace Microsoft.Quantum.Intrinsic {
     operation M (qubit : Qubit) : Result {
         return Measure([PauliZ], [qubit]);
     }
-    
-    
+
+
     /// # Summary
     /// Given a single qubit, measures it and ensures it is in the |0⟩ state
     /// such that it can be safely released.
     ///
     /// # Input
-    /// ## qubit
+    /// ## target
     /// The qubit whose state is to be reset to $\ket{0}$.
+    @RequiresCapability(
+        "BasicQuantumFunctionality",
+        "Reset is replaced by a supported implementation on all execution targets."
```

Comment:
Not for this PR, but we need a better way of handling that, since while this is true for all of the targets we provide, there is nothing that would indicate to someone defining their own target that this is required. I'll create a task to do that once we have the decompositions merged.